### ISSUE TITLE
twisted.python.zippath is inconsistent in its use of path separators

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1952]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#1952|
|Type|enhancement|
|Created|2006-07-23 08:21:04Z|

It seems like python's zipimport uses os.sep to separate internal zip paths, so twisted probably should too (except in the places where you HAVE to use "/", because that's what PKZIP uses...)

Buildbot error is here: http://twistedmatrix.com/buildbot/win32-scmikes-select/builds/55/step-default/2

<details><summary>Searchable metadata</summary>

```
trac-id__1952 1952
type__enhancement enhancement
reporter__glyph glyph
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1153642864000000 1153642864000000
changetime__1153727293000000 1153727293000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Buildbot logs are not persistent things.

Here's the output:

```
[FAIL]: twisted.test.test_paths.ZipFilePathTestCase.test_walk

Traceback (most recent call last):
  File "e:\twistedbuildbot\W32-full2.4-scmikes-select\Twisted\twisted\test\test_paths.py", line 64, in test_walk
    self.assertEquals(x, self.all)
twisted.trial.unittest.FailTest: ['e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/file1', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/sub1', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/sub1/file2', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/sub3', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/sub3/file3.ext1', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/sub3/file3.ext2', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip/sub3/file3.ext3'] != ['e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\file1', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\sub1', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\sub1\\file2', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\sub3', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\sub3\\file3.ext1', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\sub3\\file3.ext2', 'e:\\twistedbuildbot\\W32-full2.4-scmikes-select\\Twisted\\_trial_temp\\twisted.test.test_paths\\ZipFilePathTestCase\\test_walk\\bxkcny\\temp.zip\\sub3\\file3.ext3']
```

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph commented|
|-|-|

A fix for this (plus the reverted branch) is now in ```zippath-1952```.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @exarkun|
|-|-|

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @glyph|
|-|-|

Reviewed. Looks good, test that this keeps the buildslaves green (do a branch test, especially on win32).

You've omitted docstrings. Please go through and add docstrings, especially to _ZipMapImpl.mapPath and TestLoader.loadPackage.

Once you've verified that it builds, and you're satisfied that the docstring requirement of UQDS is forfilled, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set status to `closed`|
|-|-|

(In [17672]) Re-merge branch for [[#1940](https://github.com/twisted/twisted/issues/1940)](https://github.com/twisted/twisted/issues/1940), plus fix for Windows and additional docs.

Fixes [#1952](https://github.com/twisted/twisted/issues/1952)
Fixes #1940
Fixes [#10867](https://github.com/twisted/twisted/issues/10867)

Author: glyph

Reviewer: jerub

This is a do-over of r17651, hopefully this time without the test failure on
the Windows buildslave.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
